src/libparms/db/parmsdb.cpp

```diff
@@ -3371,4 +3445,77 @@ bool ParmsDb::updateLastClientSelfRestartTime(int64_t lastClientRestartTime) {
     return true;
 }
 
+
+bool ParmsDb::selectValueForKey(const std::string &key, std::string &value, const std::string &defaultValue) {
+    const std::scoped_lock lock(_mutex);
+
+    ASSERT(queryResetAndClearBindings(SELECT_VALUE_FROM_KEY_REQUEST_ID));
+    ASSERT(queryBindValue(SELECT_VALUE_FROM_KEY_REQUEST_ID, 1, key));
+    bool found = false;
+    if (!queryNext(SELECT_VALUE_FROM_KEY_REQUEST_ID, found)) {
+        LOG_WARN(_logger, "Error getting query result: " << SELECT_VALUE_FROM_KEY_REQUEST_ID);
+        value = defaultValue;
+        return false;
+    }
+    if (!found) {
+        value = defaultValue;
+        return true;
+    }
+
+    ASSERT(queryStringValue(SELECT_VALUE_FROM_KEY_REQUEST_ID, 0, value));
+    ASSERT(queryResetAndClearBindings(SELECT_VALUE_FROM_KEY_REQUEST_ID));
+
+    return true;
+}
+
+bool ParmsDb::setValueForKey(const std::string &key, const std::string &value) {
+    if (value.empty()) {
+        return deleteKeyValueEntry(key);
+    }
+
+    std::string existingValue;
+    int errId;
+    std::string error;
+    bool found = selectValueForKey(key, existingValue);
+    found = found && !existingValue.empty();
+    const std::scoped_lock lock(_mutex);
+    if (found) {  // UPDATE
+        ASSERT(queryResetAndClearBindings(UPDATE_VALUE_WITH_KEY_CLIENT_REQUEST_ID));
+        ASSERT(queryBindValue(UPDATE_VALUE_WITH_KEY_CLIENT_REQUEST_ID, 1, key));
+        ASSERT(queryBindValue(UPDATE_VALUE_WITH_KEY_CLIENT_REQUEST_ID, 2, value));
+        if (!queryExec(UPDATE_VALUE_WITH_KEY_CLIENT_REQUEST_ID, errId, error)) {
+            LOG_WARN(_logger, "Error running query: " << UPDATE_VALUE_WITH_KEY_CLIENT_REQUEST_ID);
+            return false;
+        }
+        ASSERT(queryResetAndClearBindings(UPDATE_VALUE_WITH_KEY_CLIENT_REQUEST_ID));
+    } else {  // INSERT
+        ASSERT(queryResetAndClearBindings(INSERT_KEY_VALUE_REQUEST_ID));
+        ASSERT(queryBindValue(INSERT_KEY_VALUE_REQUEST_ID, 1, key));
+        ASSERT(queryBindValue(INSERT_KEY_VALUE_REQUEST_ID, 2, value));
+
+        if (!queryExec(INSERT_KEY_VALUE_REQUEST_ID, errId, error)) {
+            LOG_WARN(_logger, "Error running query: " << INSERT_KEY_VALUE_REQUEST_ID);
+            return false;
+        }
+        ASSERT(queryResetAndClearBindings(INSERT_KEY_VALUE_REQUEST_ID));
+    }
+    return true;
+}
+
+bool ParmsDb::deleteKeyValueEntry(const std::string &key) {
```

Comment:
Should not exist.

Reply:
As suggested, I removed this function (https://github.com/Infomaniak/desktop-kDrive/pull/76/commits/2947e9f121aeeef3d61871f1601dd6cda6d61ee3). The only case I found where we might want to permanently delete an entry is during an app update when we no longer use a keyValue. In this scenario, there's nothing preventing us from implementing a similar function at that time if there are a lot of unused keyValues.